### PR TITLE
fix(dropdowns): sort category options by hierarchy label

### LINKED ISSUE #209: bug: Categories and Bank accounts on the transaction entry page are not sorted in drop downs
### What happened?

When I try to enter a transaction, I select category from the dropdown list. They are not sorted alphabetically, instead they seem to be sorted by their internal uuids.

Same applies to bank accounts I believe.

### What did you expect to happen?

Categories and Bank accounts should be sorted alphabetically in the drop downs on transaction entry page and in all other places with drop downs.

### Steps to reproduce

1. Navigate to transaction entry page
2. Click on the category dropdown list
3. Observe categories are sorted arbitrarily (not alphabetically)

### Affected area

apps/web-next

### Environment

production

### Browser / client

Chrome

### Screenshots, errors, or logs

```shell

```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iguliaev**
## Why
Issue #209 reported category and bank account dropdown options were not consistently alphabetical, and nested categories were effectively sorted by child name instead of full hierarchy label. This caused confusing ordering such as `Utilities / Heating`, `Vacation`, `Utilities / Water`.

Fixes #209.

## What Changed
Implemented parent-aware alphabetical sorting for category options and aligned dropdown ordering across key surfaces.

- Files or areas updated:
  - `apps/web-next/src/utility/categoryHierarchy.ts`
  - `apps/web-next/src/pages/transactions/{create,edit,list}.tsx`
  - `apps/web-next/src/pages/budgets/{create,edit}.tsx`
  - `apps/web-next/src/pages/categories/{create,edit}.tsx`
  - `apps/web-next/e2e/tests/transactions.spec.ts`
  - `docs/superpowers/specs/2026-07-04-issue-209-dropdown-sorting-design.md`
  - `docs/superpowers/plans/2026-07-04-issue-209-parent-aware-dropdown-sorting.md`
- New functionality added:
  - Shared category comparator that sorts by display label (`Parent / Child` when applicable)
  - New E2E regression asserting hierarchy-label ordering in transaction category dropdown
- Existing behavior changed:
  - Category dropdown ordering now follows full hierarchy label ordering
  - Transaction edit bank-account select now has explicit `name ASC` sorter

## Key Decisions
- Decision:
  - Centralize category sort behavior in shared utility helpers.
- Reasoning:
  - Keeps ordering consistent and avoids duplicated, drifting sorting logic.
- Alternatives considered:
  - Backend/view-level ordering and per-component ad hoc sorts; rejected due to broader risk or duplication.

## How This Affects the System
- User-facing changes:
  - Category dropdowns are now predictably alphabetized by full visible label.
- API changes:
  - None.
- Performance implications:
  - Minor in-memory sort of already-loaded dropdown data.
- Database changes:
  - None.
- Breaking changes:
  - None.

## Testing
- New tests added:
  - `transaction form category dropdown sorts by full hierarchy label` in `apps/web-next/e2e/tests/transactions.spec.ts`
- Existing tests status:
  - Focused transaction dropdown E2E tests passing.
- Manual testing performed:
  - N/A.
- Edge cases tested:
  - Parent/child category label ordering and parent-name search behavior in dropdown.
- Test files modified or added:
  - `apps/web-next/e2e/tests/transactions.spec.ts`

### CODE REVIEW

File: apps/web-next/src/utility/categoryHierarchy.ts [hunk 1]
```diff
@@ -31,3 +31,13 @@ export const categorySearchText = (category: Category): string => {
...
+
+/** Stable key for sorting categories by user-visible hierarchy label. */
+export const categorySortKey = (category: Category): string =>
+  categoryLabel(category).toLocaleLowerCase();
+
+/** Comparator for alphabetical ordering by full hierarchy label. */
+export const compareCategoriesByHierarchyLabel = (
+  a: Category,
+  b: Category
+): number => categorySortKey(a).localeCompare(categorySortKey(b));
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/utility/categoryHierarchy.ts]: Fixed in df41eec9. `compareCategoriesByHierarchyLabel` now uses `localeCompare(..., { sensitivity: "base" })` for consistent case-insensitive ordering across surfaces.

File: apps/web-next/src/pages/budgets/create.tsx [hunk 2]
```diff
@@ -39,13 +46,13 @@ export const BudgetCreate = () => {
 
   const categoryOptions = useMemo(
     () =>
-      categoriesQuery.data?.data
-        ?.filter((c) => !selectedType || c.type === selectedType)
-        .map((c) => ({
-          label: `${c.name} (${c.type})`,
+      (categoriesQuery.data?.data ?? [])
+        .sort(compareCategoriesByHierarchyLabel)
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/create.tsx]: Fixed in df41eec9. Updated budget create to clone the query array before sorting (`[...(categoriesQuery.data?.data ?? [])]`) to avoid in-place mutation of cached/query data.

File: apps/web-next/src/pages/budgets/edit.tsx [hunk 3]
```diff
@@ -49,13 +56,13 @@ export const BudgetEdit = () => {
 
   const categoryOptions = useMemo(
     () =>
-      categoriesQuery.data?.data
-        ?.filter((c) => !selectedType || c.type === selectedType)
-        .map((c) => ({
-          label: `${c.name} (${c.type})`,
+      (categoriesQuery.data?.data ?? [])
+        .sort(compareCategoriesByHierarchyLabel)
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/pages/budgets/edit.tsx]: Fixed in df41eec9. Updated budget edit to clone the query array before sorting (`[...(categoriesQuery.data?.data ?? [])]`) so we do not mutate React Query/Refine cached arrays.

File: apps/web-next/src/utility/categoryHierarchy.ts [hunk 4]
```diff
@@ -31,3 +31,16 @@ export const categorySearchText = (category: Category): string => {
...
+  categoryLabel(category);
+
+/** Comparator for alphabetical ordering by full hierarchy label. */
+export const compareCategoriesByHierarchyLabel = (
+  a: Category,
+  b: Category
+): number =>
+  categorySortKey(a).localeCompare(categorySortKey(b), undefined, {
+    sensitivity: "base",
+  });
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/src/utility/categoryHierarchy.ts]: Fixed in 2f96764f. Added a deterministic tie-breaker in `compareCategoriesByHierarchyLabel`: when the primary `localeCompare(..., { sensitivity: "base" })` result is 0, we now compare `id` so ordering stays stable for equivalent labels.
